skrub/_joiner.py

```diff
@@ -292,8 +287,9 @@ def fit(self, X, y=None):
             Fitted Joiner instance (self).
         """
         del y
-        X = self._check_dataframe(X)
-        self._aux_table = self._check_dataframe(self.aux_table)
+        self._aux_table = CheckInputDataFrame().fit_transform(self.aux_table)
+        self._main_check_input = CheckInputDataFrame()
+        X = self._main_check_input.fit_transform(X)
         self._check_ref_dist()
         self._check_max_dist()
         self._main_key, self._aux_key = _join_utils.check_key(
```

Comment:
we may not need all of check_key anymore, in particular left_join can now handle duplicate names in the result, but let's clean up the checks in a follow-up pr